tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/Transports/LargePayload/LargePayloadTestBase.cs

```diff
@@ -0,0 +1,109 @@
+    {
+        public LargePayloadTestBase(ITestOutputHelper output)
+            : base("LargePayload", @"test\test-applications\regression", output, prependSamplesToAppName: false)
+        {
+        }
+
+        public int FillerTagLength => 100;
+
+        public int TracesToTrigger => 100;
+
+        public int SpansPerTrace => 1999;
+
+        public int ExpectedSpans => TracesToTrigger + (TracesToTrigger * SpansPerTrace);
+
+        protected void RunTest()
+        {
+            using (var agent = EnvironmentHelper.GetMockAgent())
+            {
+                using (var sample = RunSampleAndWaitForExit(agent, arguments: $" -t {TracesToTrigger} -s {SpansPerTrace} -f {FillerTagLength}"))
+                {
+                    var spans = agent.WaitForSpans(ExpectedSpans);
+                    AssertLargePayloadExpectations(spans);
+                }
+            }
+        }
+
+        private void AssertLargePayloadExpectations(IImmutableList<MockTracerAgent.Span> spans)
+        {
+            var message = string.Empty;
+
+            var traceMap = new Dictionary<ulong, int>();
+            var spanMap = new Dictionary<ulong, int>();
+            var tagFillMap = new Dictionary<string, int>();
+
+            foreach (var span in spans)
+            {
+                var fillerGuid = span.Tags.Single(kvp => kvp.Key == "fill").Value;
+
+                if ((span.ParentId ?? 0) == 0)
+                {
+                    if (!traceMap.ContainsKey(span.TraceId))
+                    {
+                        traceMap.Add(span.TraceId, 1);
+                    }
+                    else
+                    {
+                        traceMap[span.TraceId]++;
+                    }
+                }
+
+                if (!spanMap.ContainsKey(span.SpanId))
+                {
+                    spanMap.Add(span.SpanId, 1);
+                }
+                else
+                {
+                    spanMap[span.SpanId]++;
+                }
+
+                if (!tagFillMap.ContainsKey(fillerGuid))
+                {
+                    tagFillMap.Add(fillerGuid, 1);
+                }
+                else
+                {
+                    tagFillMap[fillerGuid]++;
+                }
+            }
+
+            foreach (var duplicate in traceMap.Where(kvp => kvp.Value > 1))
+            {
+                message += $"Duplicate trace ID {duplicate.Key} with {duplicate.Value} entries {Environment.NewLine}";
+            }
+
+            foreach (var duplicate in spanMap.Where(kvp => kvp.Value > 1))
+            {
+                message += $"Duplicate span ID {duplicate.Key} with {duplicate.Value} entries {Environment.NewLine}";
+            }
+
+            foreach (var duplicate in tagFillMap.Where(kvp => kvp.Value > 1))
+            {
+                message += $"Duplicate fill ID {duplicate.Key} with {duplicate.Value} entries {Environment.NewLine}";
+            }
```

Comment:
I _think_ this is equivalent to this fluent assertions version (you'll need some `using`s too)
```suggestion
            using var scope = new AssertionsScope();
            spans.Where(x => (x.ParentId ?? 0) == 0)
                 .Should()
                 .OnlyHaveUniqueItems();

            spans.Should()
                 .OnlyHaveUniqueItems(x => x.SpanId);

            spans.Should()
                 .OnlyHaveUniqueItems(x => x.Tags.Single(kvp => kvp.Key == "fill").Value);
```